xml_converter/src/state_structs/xml_writer_state.hpp

```diff
@@ -1,7 +1,13 @@
 #pragma once
 
 #include <string>
+#include <vector>
+
+class Image;
 
 struct XMLWriterState {
-    std::string filedir;
+    const char* xml_filedir;
+
+    // A list of all of the textures with their paths.
+    std::vector<const Image*> textures;
```

Comment:
This datatype needs some more thought, it looks like you are just appending to it and then later iterating over it to copy everything out from it you might as well just copy stuff when you append to the list. I originally thought you were using this to do deduplication to prevent a texture that used by a hundred icons from being copied a hundred times.